sklearn/inspection/tests/test_h_statistic.py

```diff
@@ -0,0 +1,170 @@
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+from sklearn.base import RegressorMixin, clone
+from sklearn.datasets import make_classification, make_regression
+from sklearn.ensemble import (
+    HistGradientBoostingRegressor,
+    RandomForestClassifier,
+    RandomForestRegressor,
+)
+from sklearn.inspection import h_statistic
+from sklearn.linear_model import LinearRegression
+
+
+@pytest.mark.parametrize(
+    "est",
+    (
+        LinearRegression(),
+        HistGradientBoostingRegressor(random_state=0, max_depth=1, max_iter=20),
+    ),
+)
+@pytest.mark.parametrize("features", [None, [0, 1, 2]])
+@pytest.mark.parametrize("n_max", [500, 50])
+@pytest.mark.parametrize("sample_weight", [None, "ones"])
+def test_h_statistic_additive_regression(est, features, n_max, sample_weight):
+    # Checks that additive regressions get statistics of 0.
+    N_FEAT = 4
+    N = 200
+    X, y = make_regression(n_samples=N, n_features=N_FEAT, random_state=0)
+    w = np.ones(N) if sample_weight == "ones" else None
+
+    model = est.fit(X, y, sample_weight=w)
+    result = h_statistic(
+        model, X, features=features, sample_weight=w, n_max=n_max, random_state=1
+    )
+
+    m = N_FEAT if features is None else len(features)
+    expected_length = m * (m - 1) / 2
+
+    assert result.h_squared_pairwise.shape == (expected_length,)
+    assert_allclose(result.h_squared_pairwise, 0)
+
+
+def test_h_statistic_additive_classification():
+    # Checks that additive classification gets statistics of 0. The presence
+    # of link functions (especially for GradientBoosting with more than 2 classes)
+    # would make further tests tricky.
+    N_FEAT = 4
+    N_CLASSES = 4
+    X, y = make_classification(
+        n_samples=200,
+        n_features=N_FEAT,
+        n_informative=N_FEAT,
+        n_redundant=0,
+        n_classes=N_CLASSES,
+        random_state=0,
+    )
+    model = RandomForestClassifier(n_estimators=20, random_state=0, max_depth=1)
+    model.fit(X, y)
+    result = h_statistic(model, X, random_state=1)
+
+    expected_length = N_FEAT * (N_FEAT - 1) / 2
+
+    assert result.h_squared_pairwise.shape == (expected_length, N_CLASSES)
+    assert_allclose(result.h_squared_pairwise, 0)
+
+
+@pytest.mark.parametrize(
+    "est",
+    (
+        RandomForestRegressor(random_state=0, max_depth=4, n_estimators=20),
+        HistGradientBoostingRegressor(random_state=0, max_iter=20),
+    ),
+)
+@pytest.mark.parametrize("features", [None, [0, 1, 2]])
+@pytest.mark.parametrize("n_max", [500, 50])
+@pytest.mark.parametrize("sample_weight", [None, "ones"])
+def test_h_statistic_regression(est, features, n_max, sample_weight):
```

Comment:
Could this test be merged with test_h_statistic_additive_regression?
Something like:
```python

@pytest.mark.parametrize(
    ["est", "is_additive",
    (
        [LinearRegression(), True],
        [HistGradientBoostingRegressor(random_state=0, max_depth=1, max_iter=20), True],
        [RandomForestRegressor(random_state=0, max_depth=4, n_estimators=20), False],
        [HistGradientBoostingRegressor(random_state=0, max_iter=20), False],
    ),
)

...

    if is_additive:
        assert_allclose(result.h_squared_pairwise, 0)
    else:
        assert any(result.h_squared_pairwise > 1e-5)
```